File: bin/_template.py
#!/usr/bin/env python

"""
A Python script template

Usage:
    template.py --out-file=<path> [options] <file>

Options:
    -h --help            Show this screen.
    --out-file=<path>    Path to output file.
"""


def run_analysis(input):
    """
    A function that performs analysis

    Parameters
    ----------
    input
        The function input

    Returns
    -------
    The function output
    """

    output = input

    return output


def main():
    """The main script function"""
    from docopt import docopt

    args = docopt(__doc__)

    file = args["<file>"]
    out_file = args["--out-file"]

    print(f"Reading data from '{file}'...")
    # input = read_data(file)
    print("Read data:")
    # print(input)
    output = run_analysis(input)
    print(f"Writing output to '{out_file}'...")
    # write_output(output, out_file)
    print("Done!")


if __name__ == "__main__":
    main()

File: bin/_template.R
#!/usr/bin/env Rscript

"
An R script template

Usage:
    template.R --out-file=<path> [options] <file>
Options:
    -h --help             Show this screen.
    --out-file=<path>     Path to output file.
" -> doc

# Load libraries
suppressPackageStartupMessages({

})

#' A function that performs analysis
#'
#' @param input The function input
#'
#' @returns The function output
run_analysis <- function(input) {

    output = input

    return(input)
}

#' The main script function
main <- function() {
    args <- docopt::docopt(doc)

    file      <- args[["<file>"]]
    out_file  <- args[["--out-file"]]

    message("Reading data from '", file, "'...")
    # input <- read_data(file)
    message("Read data:")
    # print(input)
    output <- run_analysis(input)
    message("Writing output to '", out_file, "'...")
    # write_output(output, out_file)
    message("Done!")
}

if (sys.nframe() == 0) {
    main()
}
